### ISSUE TITLE
Construct 3 game not working correctly due to "403 forbidden" fetch request

### ISSUE THREAD

**Issue by F3der1co** [itchio/itch.io#1383]
_I reported this bug to Scirra (the developers of Construct) but the answer was that this needs to be fixed by itch.io:
https://github.com/Scirra/Construct-3-bugs/issues/6223_

A game has some bugs when I run it on itch.io but it works on other platforms.

**It seems like itch.io is not allowing a worker request. (afaik c3 runs certain logic on a worker for multi-threadting purposes)**

![errorWithWorker](https://user-images.githubusercontent.com/89512047/199001110-5f9a3167-77be-408e-96bc-72a3850b9a0e.png)

**Is there a reason this request is forbidden?**

**Comment by leafo** [comment]
Please read the HTML guide, specifically the common pitfalls section: https://itch.io/docs/creators/html5#common-pitfalls

In your code you are making a request to something that doesn't exist. Because of our bucket's security model 403 is returned instead of the commonly seen 404 error.

**Comment by F3der1co** [comment]
so tg

> Please read the HTML guide, specifically the common pitfalls section: https://itch.io/docs/creators/html5#common-pitfalls
> 
> In your code you are making a request to something that isn't a file (in this case, attempting to fetch a directory will always fail, you can only fetch valid files directly by name). Because of our bucket's security model 403 is returned in cases where you might expect to see a 404 error. It's likely you have misconfigured your build or project settings.
> 
> Hope that helps

This is an empty project I am uploading.
So this means games making use of a web worker are not supported on itch/ you are not interested in supporting them?

**Comment by leafo** [comment]
> So this means games making use of a web worker are not supported on itch/ you are not interested in supporting them?

No, please read the error message in the browser console. We run the code you provide exactly as is, if you upload something that is broken then it's not going to work.

Your service worker is making a request to the URL: `https://v6p9d9t4.ssl.hwcdn.net/html/6735731/`. This will always fail since it's not a file, but the path to a directory. Only *files* that you have provided in your zip are valid resources to be fetched. To fix the issue, change it such that your service working is not trying to request the URL of a resource that does not exist.

> This is an empty project I am uploading.

Even empty projects can have configuration, hence my suggestions I originally wrote: *It's likely you have misconfigured your build or project settings.*

Hope that helps

**Comment by F3der1co** [comment]
thanks for the quick answer, I'm pretty sure that any c3 project does this request no matter the export settings. I will contact Scirra again with the info you provided to see if they have an idea.

**Comment by AshleyScirra** [comment]
Construct's Service Worker makes this request as part of its offline caching. Construct does not know in advance how the web server is configured. Caching the main document is tricky - there are, for example, patterns where a user loads the game by visiting a URL like `example.com/index.html`, goes offline, and then visits `example.com/`, and expects the game to load. In this case there are actually two main documents: `index.html` and then the root path `/` which usually defaults to serving index.html as well. This is the kind of thing that has come from battle-testing web exports on a wide range of different kinds of servers over a period of years.

It looks like itch.io instead serves a 403 Forbidden on the root `/` path, which is unfortunate as in this situation Construct gets an error trying to cache the main document there and so determines offline caching has failed. It would be useful in this case if itch.io could just serve or redirect to `index.html` at that path instead of returning an error. I guess we could try to make Construct's SW error tolerant in this case, but coding service workers is generally a nightmare, and this would improve itch.io compatibility with the majority of other kinds of web servers I've come across. Could itch.io consider making that change?
